syntax = "proto3";

package aimodelservice;

message TelegramMessage {
  string symbol = 1;
  string exchange = 2; // Binance / Bybit
  repeated double trigger_features = 3; // null заменим на 0 в Python
  int64 timestamp = 4;
}

message MarketDataMessage {
  string symbol = 1;
  repeated double ohlcv_1m = 3;
  repeated double ohlcv_5m = 4;
  double open_interest = 5;
  double ma163 = 6;
  double ma200 = 7;
  double ma360 = 8;
  double vwap = 9;
  double vwap_dev1 = 10;
  double vwap_dev2 = 11;
  double cvd_spot = 12;
  double cvd_futures = 13;
  double liq_buy = 14;
  double liq_sell = 15;
  int64 timestamp = 16;
}

message PredictionResponse {
  string symbol = 1;
  double action = 2; // 1 = buy, -1 = sell, 0 = hold
  double confidence = 3;
}

service AIModelService{
  rpc SendTrigger (TelegramMessage) returns (PredictionResponse);
  rpc SendMarketData (MarketDataMessage) returns (PredictionResponse);
}
